### DISCUSSION TITLE
Can I use databend without copying data?

### DISCUSSION THREAD

**Discussion by kesavkolla** [databendlabs/databend#8842] [Q&A] [answered]
I have very large dataset already in s3 (parquet files). I don't want to copy that data again into databend. Is it possible to use databend query engine to run queries against data which is s3?

**Comment by bohutang** [comment, marked as answer]
Not yet, we are working on it, tracking issue: https://github.com/datafuselabs/databend/issues/7211

**Comment by kesavkolla** [comment]
In that case can fuse engine be accessible via arrow flight? Would be nice to run query engine via databend and the results are sent via arrow flight.

**Reply by bohutang** [reply]
That's easy to do. 
Now Databend supports `HTTP/MySQL` handler, we add an Arrow flight handler https://github.com/datafuselabs/databend/tree/main/src/query/service/src/servers will do it.

**Reply by sundy-li** [reply]
https://github.com/datafuselabs/databend/issues/8857

**Comment by dpy1123** [comment]
@kesavkolla u can use apache drill to directly query from ur exsiting s3 files, but drill for s3 does not have query index.

**Reply by kesavkolla** [reply]
I don't think drill can leverage apache arrow for doing native executions. There is lot of value in arrow and it's native expression engine.

**Comment by gianluca-moro** [comment]
Any updates on this? I have the same situation: Large files in s3 which I want to directly query.

**Reply by bohutang** [reply]
Yes, Databend can query data from parquet format file, but it will slower than FUSE engine, doc is here:
https://databend.rs/doc/sql-functions/table-functions/stage_table_function#query-examples

**Reply by gianluca-moro** [reply]
Is it possible to create a `view` on such a `select * from 's3://bucket/'` query?

**Reply by bohutang** [reply]
Yes

**Reply by gianluca-moro** [reply]
I tried the following:
```
CREATE VIEW region_view AS 
SELECT 
    r_regionkey,
    r_name,
    r_comment
FROM 's3://my_bucket/region/'
(
 access_key_id => '<my_access_key>', 
 secret_access_key => '<my_secret_key>',
 region => 'eu-central-1',
 file_format => 'parquet',
 pattern => '.*parquet$'
);
```
However, this fails with `APIError: ResponseError with 1025: VIEW QUERY: system.read_parquet not exists`
